plugins/postjobexecutionhooks/pushtoS3/s3.go

```diff
@@ -22,243 +23,163 @@ import (
-							}
-							msg := "Error in commit " + jobSha + ". Something goes wrong in the test with the jobName: " + jobName
-							err = slackAPI.MsgToSlack(msg)
-							if err != nil {
-								fmt.Println("Error could not posted to slack: error", err)
-							}
-						} else if matchsucceed && !matcherr {
-							for _, teststatus := range binaryURL {
-								if teststatus.TestName == "push coreboot binary to S3" {
-									var TestStepStatuses = teststatus.TestStepStatuses
-									for _, teststepstatus := range TestStepStatuses {
-										if teststepstatus.TestStepCoordinates.TestStepName == "UploadFile" {
-											var TargetStatuses = teststepstatus.TargetStatuses
-											for _, targetstatus := range TargetStatuses {
-												for _, events := range targetstatus.Events {
-													if events.Data.EventName == "CmdStdout" {
-														var url url
-														var status_desc = jobName + ". Binary in S3 Bucket:"
-														json.Unmarshal(*events.Data.Payload, &url)
-														err := githubAPI.EditGithubStatus(ctx, "success", url.Msg, status_desc, jobSha)
-														if err != nil {
-															fmt.Println("GithubStatus could not be edited to status: success", err)
-														}
-													}
-												}
-											}
-										}
-									}
-								}
-							}
-							err := githubAPI.EditGithubStatus(ctx, "success", fileurl, status_desc, jobSha)
-							if err != nil {
-								fmt.Println("GithubStatus could not be edited to status: error", err)
-							}
-							msg := "The test with the jobName '" + jobName + "' was successful."
-							err = slackAPI.MsgToSlack(msg)
-							if err != nil {
-								fmt.Println("Error could not posted to slack: error", err)
-							}
-						}
-					}
+		// Creating link where the job report can be downloaded.
+		// This link will be put into the commit message right after the test status
+		resultURL := strings.Join([]string{"https://", parameter.s3Bucket, ".s3.", parameter.s3Region, ".amazonaws.com/", uploadPath}, "")
+
+		// Go through the report and retrieve the job status
+		jobStatus = statusResp.Data.Status.JobReport.RunReports
+
+		// Go through all final reports
+		jobSuccess := true
+		for _, allReports := range jobStatus {
+			//Go through all report in the final reports
+			for _, reports := range allReports {
+				var success = reports.Success
+				if !success {
+					jobSuccess = false
 				}
-				return nil
 			}
+		}
+		// Adapt the Github Commit statuses and Slack Msg depending on the job success
+		// Creating the description for the result status and for the binary status
+		resultDesc := jobName + ". Test-Result:"
+		binaryDesc := jobName + ". Binary in S3 Bucket:"
+		// Parse the status resp for the binary url to update the binary status
+		regex := "https://" + parameter.s3Bucket + `[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)`
```

Comment:
I don't know exactly what you mean.

Reply:
Isn't the upload path fixed? So can't this be included?

Reply:
No the upload path in the contest-client is referring to the test results, because it uploads it. The path that is in the URL is the one that is referring to the binaries.